### ISSUE TITLE
Basic CRUD component example.

### ISSUE THREAD

**Issue by andrew-za** [rappasoft/laravel-5-boilerplate#543]
Thanks for all the hard work put into this project!

Would like to suggest packaging a basic CRUD component. For new people to this framework we resort to copying and studying an existing component in order to figure out how everything works.

Now I managed todo that with the Users component - but like someone else mentioned it is a more complex component with UserRoles integration etc require that code be stripped out.

Preferably without the dataTable library - I understand that it does allot but it feels restrictive and do not like the second delay waiting for he request to fetch data.

**Comment by mpixelz** [comment]
i agree.. ive been using laravel backpack but now wanted to try this boilerplate but it seems too complicated to me for now.. totally different than what ive used before.
it would be really great if there are any examples so we can study them..
for instance i tried to replicate user list view but apply filter to show only executive role users.. but cant figure out how to.

**Comment by rappasoft** [comment]
It's really not hard to follow the code at all if you've read the Laravel documentation which I suggest you do so, there's not that much custom stuff on top of the default installation. Most of it is just convenience.

**Comment by derekphilipau** [comment]
This project is perfect for my needs while rebuilding a site from Laravel 5.1.  I agree with rappasoft, if you have a foundation with Laravel it shouldn't be difficult to get going.  However, I've found there's a steep learning curve with Vue 2.0, most of the examples I've come across on Laracasts and other websites are older versions of Vue.  It would be nice to see an example of how Vue should be used rather than simply including it.  The project https://github.com/fagray/vuejs2-laravel53-starter has a working example, although I'm not sure if it follows best practices.

**Comment by derekphilipau** [comment]
This project is perfect for my needs while rebuilding a site from Laravel 5.1. I agree with rappasoft, if you have a foundation with Laravel it shouldn't be difficult to get going. However, I've found there's a steep learning curve with Vue 2.0, most of the examples I've come across on Laracasts and other websites are older versions of Vue. It would be nice to see an example of how Vue _should_ be used, perhaps in the backend. The project https://github.com/fagray/vuejs2-laravel53-starter has a working example, although I'm still learning and not sure if it follows best practices.

**Comment by blomdahldaniel** [comment]
I agree as well, it might feel hard to hear but with knowledge of Laravel fundamentals this boilerplate should make sense. This boilerplate cannot supply resources to teach Laravel basics. 

@derekphilipau really nice boilerplate for Vue 2.0! Glad to see it! Have been looking for something lika this myself.

I will close this issue for now.

**Comment by andrew-za** [comment]
I completely understand your reasons for NOT doing this. I just think its a great opportunity to teach people the right way todo things.
I struggling learning from reading documentation - goes much quicker for me by looking at coding examples. Most tutorials and help has examples written for basic applications the logic is not so neat and well structured as it is with this boiler plate.

For example - I am now trying to figure out where to put the logic to handle image upload using the Intervention Image library. One recommends helper method, another a service provider.

I just think a basic CRUD application will help a real bunch.

**Comment by blomdahldaniel** [comment]
I feel you man! Learning is hard, especially when you have to do it on your own. I have learned all my Laravel-knowledge mainly from [Laracasts](https://laracasts.com/). I would strongly recommend going through some stuff there.
- [Laravel 5 Fundamentals](https://laracasts.com/series/laravel-5-fundamentals)
- [Laravel Intermidiate](https://laracasts.com/series/intermediate-laravel)

2 good courses well worth the money in my opinion.

If all you are looking for is a CRUD backend I can recommend you another laravel boilerplate. |[Voyager] (https://github.com/the-control-group/voyager) they focus mainly on an entire CRUD-system. A cool aspect of how to do things. There you will find endless of cool ways to manage CRUD.

**Comment by andrew-za** [comment]
@blomdahldaniel Thanks that is indeed a great library. My gut tells me to stay with this boilerplate and learn rather than getting spoon fed ;-)
